File: backend/src/main/java/com/healthcare/dashboard/entities/Prediction.java
package com.healthcare.dashboard.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "predictions")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Prediction {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String type; // REVENUS, PATIENTS, OCCUPATION, COUTS, TENDANCES
    
    @Column(nullable = false)
    private String titre;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(nullable = false)
    private LocalDateTime periodePrevue;
    
    @Column(columnDefinition = "TEXT")
    private String donneesHistoriques; // JSON format
    
    @Column(columnDefinition = "TEXT")
    private String resultatPrediction; // JSON format avec valeurs prédites
    
    private Double confiance; // Niveau de confiance en pourcentage
    
    @Column(columnDefinition = "TEXT")
    private String methodologie; // Description de l'algorithme utilisé
    
    @Column(columnDefinition = "TEXT")
    private String facteursCles;
    
    @Column(columnDefinition = "TEXT")
    private String recommandations;
    
    private String generePar; // AI ou utilisateur
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: flutter_app/lib/models/prediction.dart
class Prediction {
  final int? id;
  final String type;
  final String titre;
  final String? description;
  final String periodePrevue;
  final String? donneesHistoriques;
  final String? resultatPrediction;
  final double? confiance;
  final String? methodologie;
  final String? facteursCles;
  final String? recommandations;
  final String? generePar;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Prediction({
    this.id,
    required this.type,
    required this.titre,
    this.description,
    required this.periodePrevue,
    this.donneesHistoriques,
    this.resultatPrediction,
    this.confiance,
    this.methodologie,
    this.facteursCles,
    this.recommandations,
    this.generePar,
    this.createdAt,
    this.updatedAt,
  });

  factory Prediction.fromJson(Map<String, dynamic> json) {
    return Prediction(
      id: json['id'],
      type: json['type'] ?? '',
      titre: json['titre'] ?? '',
      description: json['description'],
      periodePrevue: json['periodePrevue'] ?? '',
      donneesHistoriques: json['donneesHistoriques'],
      resultatPrediction: json['resultatPrediction'],
      confiance: json['confiance']?.toDouble(),
      methodologie: json['methodologie'],
      facteursCles: json['facteursCles'],
      recommandations: json['recommandations'],
      generePar: json['generePar'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'type': type,
      'titre': titre,
      if (description != null) 'description': description,
      'periodePrevue': periodePrevue,
      if (donneesHistoriques != null) 'donneesHistoriques': donneesHistoriques,
      if (resultatPrediction != null) 'resultatPrediction': resultatPrediction,
      if (confiance != null) 'confiance': confiance,
      if (methodologie != null) 'methodologie': methodologie,
      if (facteursCles != null) 'facteursCles': facteursCles,
      if (recommandations != null) 'recommandations': recommandations,
      if (generePar != null) 'generePar': generePar,
    };
  }
}
